EOF behavior

When reading the entire contents in a single `Read` call, BrotliReader returns `io.EOF`.

This differs from a `bytes.Reader` for example, see this self-contained test:

```go
package pwr_test

import (
	"bytes"
	"io"
	"testing"

	"gopkg.in/kothar/brotli-go.v0/dec"
	"gopkg.in/kothar/brotli-go.v0/enc"
)

func must(t *testing.T, err error) {
	if err != nil {
		t.Fatal(err.Error())
		t.FailNow()
	}
}

func makeBrotliReader(t *testing.T, data []byte) io.Reader {
	params := enc.NewBrotliParams()
	params.SetQuality(1)

	buf := new(bytes.Buffer)
	writer := enc.NewBrotliWriter(params, buf)
	_, err := writer.Write(data)
	must(t, err)

	err = writer.Close()
	must(t, err)

	return dec.NewBrotliReader(bytes.NewReader(buf.Bytes()))
}

func Test_BrotliEOFBehavior(t *testing.T) {
	buf := []byte{0x1, 0x2, 0x3, 0x4}

	normalReader := bytes.NewReader(buf)
	brotliReader := makeBrotliReader(t, buf)

	receiver := make([]byte, len(buf))

	_, err := normalReader.Read(receiver)
	t.Log("Normal read:")
	must(t, err)

	_, err = brotliReader.Read(receiver)
	t.Log("Brotli read:")
	must(t, err)
}
```

The results are: 

```
$ go test -v ./pwr/brotli_eof_test.go
=== RUN   Test_BrotliEOFBehavior
--- FAIL: Test_BrotliEOFBehavior (0.07s)
        brotli_eof_test.go:43: Normal read:
        brotli_eof_test.go:47: Brotli read:
        brotli_eof_test.go:14: EOF
FAIL
exit status 1
```

--- a/dec/decode.go
+++ b/dec/decode.go
@@ -166,6 +166,9 @@ func (r *BrotliReader) Read(p []byte) (n int, err error) {
 		}
 	}
 
+	if r.err == io.EOF && n > 0 {
+		return n, nil
+	}
 	return n, r.err
 }
 

--- a/dec/decode_test.go
+++ b/dec/decode_test.go
@@ -112,3 +112,29 @@ func TestGCErrors(T *testing.T) {
 		}
 	}
 }
+
+func TestEOFBehavior(T *testing.T) {
+	input := []byte{1, 2, 3, 4}
+	output := make([]byte, len(input)*12)
+
+	params := enc.NewBrotliParams()
+	params.SetQuality(4)
+
+	output, err := enc.CompressBuffer(params, input, output)
+	if err != nil {
+		T.Fatal(err)
+	}
+
+	// Decompress as a stream
+	reader := NewBrotliReader(bytes.NewReader(output))
+	receiver := make([]byte, len(input))
+
+	readBytes, err := reader.Read(receiver)
+	if err != nil {
+		T.Fatal(err)
+	}
+
+	if readBytes != len(input) {
+		T.Fatalf("Expected to read %d bytes, read %d", len(input), readBytes)
+	}
+}

